src/pip/_internal/cli/autocompletion.py

```diff
@@ -32,7 +35,7 @@ def autocomplete():
     try:
         subcommand_name = [w for w in cwords if w in subcommands][0]
     except IndexError:
-        subcommand_name = None
+        subcommand_name = None  # type: ignore
```

Comment:
I would put `# type: Optional[List[str]]` on the first assignment instead.

Reply:
Or we can just change this line to `subcommand_name = ''`.

Reply:
Indeed, thanks=) I've rewritten the piece to get rid of try/except.